building_sim_plugins/building_plugins_common/include/building_sim_common/crowd_simulator_common.hpp

```diff
@@ -238,132 +246,202 @@ template<typename SdfPtrT>
 
-    if (!model_type_element->template Get<double>("animation_speed",
-      model_type_ptr->animation_speed, 0.0))
+    if (!sdf->template HasElement("model_type"))
     {
-      RCLCPP_ERROR(
-        logger(),
-        "No animation speed configured in <model_type>! <animation_speed> Required");
+      RCLCPP_ERROR(logger(),
+        "No model type for agents found! <model_type> element Required!");
       return false;
     }
+    auto model_type_element = sdf->template GetElementImpl("model_type");
+    while (model_type_element)
+    {
+      std::string s;
+      if (!model_type_element->template Get<std::string>("typename", s, ""))
+      {
+        RCLCPP_ERROR(logger(),
+          "No model type name configured in <model_type>! <typename> Required");
+        return false;
+      }
+
+      auto model_type_ptr = this->_model_type_db_ptr->emplace(s,
+          std::make_shared<ModelTypeDatabase::Record>() ); //unordered_map
+      model_type_ptr->type_name = s;
+
+      if (!model_type_element->template Get<std::string>("animation",
+        model_type_ptr->animation, ""))
+      {
+        RCLCPP_ERROR(logger(),
+          "No animation configured in <model_type>! <animation> Required");
+        return false;
+      }
+
+      if (!model_type_element->template Get<double>("animation_speed",
+        model_type_ptr->animation_speed, 0.0))
+      {
+        RCLCPP_ERROR(
+          logger(),
+          "No animation speed configured in <model_type>! <animation_speed> Required");
+        return false;
+      }
+
+      model_type_element = model_type_element->template GetNextElement(
+        "model_type");
+    }
 
-    if (!model_type_element->template HasElement("initial_pose"))
+    if (!sdf->template HasElement("external_agent"))
     {
       RCLCPP_ERROR(
         logger(),
-        "No model initial pose configured in <model_type>! <initial_pose> Required [" + s +
-        "]");
-      return false;
+        "No external agent provided. <external_agent> is needed with a unique name defined above.");
+    }
+    auto external_agent_element =
+      sdf->template GetElementImpl("external_agent");
+    while (external_agent_element)
+    {
+      auto ex_agent_name = external_agent_element->template Get<std::string>();
+      RCLCPP_INFO(logger(),
+        "Added external agent: [ " + ex_agent_name + " ].");
+      _external_agents.emplace_back(ex_agent_name); //just store the name
+      external_agent_element = external_agent_element->template GetNextElement(
+        "external_agent");
     }
-    if (!_load_model_init_pose(model_type_element, model_type_ptr->pose))
+    if (!sdf->template HasElement("internal_agent"))
     {
       RCLCPP_ERROR(
         logger(),
-        "Error loading model initial pose in <model_type>! Check <initial_pose> in [" + s +
-        "]");
-      return false;
+        "No internal agent provided. <internal_agent> is needed with a unique name defined above.");
```

Comment:
Same as the external agent error, this would be a simulation without humans? Also what does "defined above" mean?